frame/staking/src/lib.rs

```diff
@@ -529,13 +529,26 @@ impl<T: Config> StakingLedger<T> {
 		(self, unlocking_balance)
 	}
 
-	/// Slash the staker for a given amount of balance. This can grow the value of the slash in the
-	/// case that either the active bonded or some unlocking chunks become dust after slashing.
-	/// Returns the amount of funds actually slashed.
+	/// Slash the staker for a given amount of balance.
 	///
-	/// `slash_era` is the era in which the slash (which is being enacted now) actually happened.
+	/// This implements a proportional slashing system, whereby we set our preference to slash as
+	/// such:
+	///
+	/// - If any unlocking chunks exist that are scheduled to be unlocked at `slash_era +
+	///   bonding_duration` and onwards, the slash is divided equally between the active ledger and
+	///   the unlocking chunks.
```

Comment:
One question I have is why we divide the slash in this way, rather than e.g. slashing first out of the unlocking chunks and if we need to slash more, start slashing the active ledger. I.e. what is the motivation for wanting a *proportional* slash?

Reply:
This is exactly what/how the old behavior was, now we do it this way so that nomination pools become more fair.

For example, in a pool, we want all members who are active in a pool, or unbonded in the last few eras to equally be affected by the slash, not just the active ones.